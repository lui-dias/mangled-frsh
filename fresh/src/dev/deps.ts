// std
export {
  basename,
  dirname,
  extname,
  fromFileUrl,
  join,
  posix,
  relative,
  resolve,
  SEP,
  toFileUrl,
} from "https://deno.land/std@0.193.0/path/mod.ts";
export { DAY, WEEK } from "https://deno.land/std@0.193.0/datetime/constants.ts";
export * as colors from "https://deno.land/std@0.193.0/fmt/colors.ts";
export {
  walk,
  type WalkEntry,
  WalkError,
} from "https://deno.land/std@0.193.0/fs/walk.ts";
export { parse } from "https://deno.land/std@0.193.0/flags/mod.ts";
export { gte } from "https://deno.land/std@0.193.0/semver/mod.ts";
export { existsSync } from "https://deno.land/std@0.193.0/fs/mod.ts";
export * as semver from "https://deno.land/std@0.195.0/semver/mod.ts";
export * as JSONC from "https://deno.land/std@0.195.0/jsonc/mod.ts";
export * as fs from "https://deno.land/std@0.195.0/fs/mod.ts";

// ts-morph
export { Node, Project } from "https://deno.land/x/ts_morph@17.0.1/mod.ts";
