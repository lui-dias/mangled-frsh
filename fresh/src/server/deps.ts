// -- std --
export {
  dirname,
  extname,
  fromFileUrl,
  isAbsolute,
  join,
  toFileUrl,
} from "https://deno.land/std@0.193.0/path/mod.ts";
export { walk } from "https://deno.land/std@0.193.0/fs/walk.ts";
export * as colors from "https://deno.land/std@0.193.0/fmt/colors.ts";
export {
  type Handler as ServeHandler,
  serve,
} from "https://deno.land/std@0.193.0/http/server.ts";
export { Status } from "https://deno.land/std@0.193.0/http/http_status.ts";
export {
  typeByExtension,
} from "https://deno.land/std@0.193.0/media_types/mod.ts";
export { toHashString } from "https://deno.land/std@0.193.0/crypto/to_hash_string.ts";
export { escape } from "https://deno.land/std@0.193.0/regexp/escape.ts";
export * as JSONC from "https://deno.land/std@0.193.0/jsonc/mod.ts";
