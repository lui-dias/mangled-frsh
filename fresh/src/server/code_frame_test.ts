import { assertEquals } from "$std/testing/asserts.ts";
import { assertSnapshot } from "$std/testing/snapshot.ts";
import { createCodeFrame, getFirstUserFile } from "./code_frame.ts";
import { colors } from "./deps.ts";

function testCodeFrame(text: string, line: number, column: number) {
  const codeFrame = createCodeFrame(text, line, column);
  if (codeFrame !== undefined) {
    return "\n" + colors.stripColor(codeFrame);
  }
  return codeFrame;
}

const FILE = `file://example.com/foo.js`;

Deno.test("getFirstUserFile", () => {
  let stack = `Some message
  at asdf.foo (https://example.com/foo.js:8:20)
  at asdf.foo (${FILE}:8:20)
  `;
  assertEquals(getFirstUserFile(stack), {
    fnName: "asdf.foo",
    file: FILE,
    column: 20,
    line: 8,
  });

  stack = `Some message
  at asdf.foo (https://example.com/foo.js:8:20)
  at (${FILE}:8:20)
  `;
  assertEquals(getFirstUserFile(stack), {
    fnName: "",
    file: FILE,
    column: 20,
    line: 8,
  });
});

Deno.test("createCodeFrame invalid ranges", () => {
  assertEquals(createCodeFrame("", 2, 10), undefined);
  assertEquals(createCodeFrame("asdf", 2, 10), undefined);
  assertEquals(createCodeFrame("asdf", 1, 10), undefined);
});

Deno.test("createCodeFrame markers", async (t) => {
  await assertSnapshot(t, testCodeFrame("foo\nbar\nbaz", 1, 2));
  await assertSnapshot(t, testCodeFrame("foo\nbar\nbaz", 0, 0));
  await assertSnapshot(t, testCodeFrame("foo\nbar\nbaz", 2, 2));
  await assertSnapshot(
    t,
    testCodeFrame("foo\n\n\n\n\n\n\n\nbar\n\n\nbar", 8, 0),
  );
});

Deno.test("createCodeFrame with indentation", async (t) => {
  await assertSnapshot(t, testCodeFrame("foo\n      bar\nbaz", 1, 6));
  await assertSnapshot(t, testCodeFrame("foo\n\t\tbar\nbaz", 1, 2));
});
