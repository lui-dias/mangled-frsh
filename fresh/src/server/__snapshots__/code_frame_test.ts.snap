export const snapshot = {};

snapshot[`createCodeFrame markers 1`] = `
"
  1 | foo
> 2 | bar
    |   ^
  3 | baz
"
`;

snapshot[`createCodeFrame markers 2`] = `
"
> 1 | foo
    | ^
  2 | bar
  3 | baz
"
`;

snapshot[`createCodeFrame markers 3`] = `
"
  1 | foo
  2 | bar
> 3 | baz
    |   ^
"
`;

snapshot[`createCodeFrame markers 4`] = `
"
   7 | 
   8 | 
>  9 | bar
     | ^
  10 | 
  11 | 
  12 | bar
"
`;

snapshot[`createCodeFrame with indentation 1`] = `
"
  1 | foo
> 2 |       bar
    |       ^
  3 | baz
"
`;

snapshot[`createCodeFrame with indentation 2`] = `
"
  1 | foo
> 2 |     bar
    |     ^
  3 | baz
"
`;
